app/controllers/charge_controller.js

```diff
@@ -129,81 +127,72 @@ module.exports = {
 
     if (charge.status === State.AUTH_READY) return redirect(res).toAuthWaiting(req.chargeId)
     // else
-    AWSXRay.captureAsyncFunc('chargeValidator_verify', function (subSegment) {
-      validator.verify(req)
-        .catch(() => {
-          subSegment.close('error')
-          redirect(res).toNew(req.chargeId)
-        })
-        .then(data => {
-          subSegment.close()
-          card = data.card
+    validator.verify(req)
+      .catch(() => {
+        redirect(res).toNew(req.chargeId)
+      })
+      .then(async data => {
+        card = data.card
 
-          let emailTypos = false
-          let emailPatch
-          let userEmail
+        let emailTypos = false
+        let userEmail
 
-          if (
-            charge.gatewayAccount.emailCollectionMode === 'OFF' ||
-            (charge.gatewayAccount.emailCollectionMode === 'OPTIONAL' && (!req.body.email || req.body.email === ''))
-          ) {
-            emailPatch = Promise.resolve('Charge patch skipped as email collection mode was toggled off, or optional and not supplied')
-          } else {
-            userEmail = req.body.email
-            let emailChanged = false
-            if (req.body.originalemail) {
-              emailChanged = req.body.originalemail !== userEmail
-            }
-            emailTypos = commonTypos(userEmail)
-            if (req.body['email-typo-sugestion']) {
-              userEmail = emailChanged ? req.body.email : req.body['email-typo-sugestion']
-              emailTypos = req.body['email-typo-sugestion'] !== req.body.originalemail ? commonTypos(userEmail) : null
-            }
-            emailPatch = Charge(req.headers[CORRELATION_HEADER]).patch(req.chargeId, 'replace', 'email', userEmail, subSegment)
+        if (charge.gatewayAccount.emailCollectionMode !== 'OFF' ||
+          (charge.gatewayAccount.emailCollectionMode !== 'OPTIONAL' && req.body.email && req.body.email !== '')
```

Comment:
An empty string is a falsy value, so we can get rid of `req.body.email !== ''` I think

Reply:
Also not sure this conditional is quite right, it will always return true unless collection more is 'OFF'.
Maybe it should check if collection mode is 'ON' or 'OPTIONAL' and email is provided.

Reply:
fixed